plasmapy/particles/tests/test_particle_collections.py

```diff
@@ -322,3 +326,126 @@ def test_particle_multiplication(method, particle):
     """
     particle_list = getattr(particle, method)(3)
     assert particle_list == [particle, particle, particle]
+
+
+def test_mean_particle():
+    """
+    Test that ``ParticleList.average_particle()`` returns a particle with
+    the mean mass and mean charge of a |ParticleList|.
+    """
+    massless_uncharged_particle = CustomParticle(mass=0 * u.kg, charge=0 * u.C)
+    particle_list = ParticleList([proton, electron, alpha, massless_uncharged_particle])
+    expected_mass = (proton.mass + electron.mass + alpha.mass) / 4
+    expected_charge = (proton.charge + electron.charge + alpha.charge) / 4
+    average_particle = particle_list.average_particle()
+    assert u.isclose(average_particle.mass, expected_mass, rtol=1e-14)
+    assert u.isclose(average_particle.charge, expected_charge, rtol=1e-14)
+
+
+def test_weighted_mean_particle():
+    """
+    Test that ``ParticleList.average_particle`` returns a particle with
+    the weighted mean.
+    """
+    custom_proton = CustomParticle(mass=proton.mass, charge=proton.charge)
+    particle_list = ParticleList([proton, electron, alpha, custom_proton])
+    abundances = [1, 2, 0, 1]
+    expected_mass = (proton.mass + electron.mass) / 2
+    expected_charge = 0 * u.C
+    average_particle = particle_list.average_particle(abundances=abundances)
+    assert u.isclose(average_particle.mass, expected_mass, rtol=1e-14)
+    assert u.isclose(average_particle.charge, expected_charge, rtol=1e-14)
+
+
+boolean_pairs = [(False, False), (True, False), (False, True), (True, True)]
+
+
+@pytest.mark.parametrize("use_rms_charge, use_rms_mass", boolean_pairs)
+def test_root_mean_square_particle(use_rms_charge, use_rms_mass):
+    """
+    Test that ``ParticleList.average_particle`` returns the mean or root
+    mean square of the charge and mass, as appropriate.
+    """
+
+    particle_list = ParticleList(["p+", "e-"])
+    average_particle = particle_list.average_particle(
+        use_rms_charge=use_rms_charge, use_rms_mass=use_rms_mass
+    )
+
+    expected_average_charge = (1 if use_rms_charge else 0) * proton.charge
+    assert u.isclose(average_particle.charge, expected_average_charge, rtol=1e-14)
+
+    if use_rms_mass:
+        expected_average_mass = np.sqrt((proton.mass ** 2 + electron.mass ** 2) / 2)
+    else:
+        expected_average_mass = (proton.mass + electron.mass) / 2
+
+    assert u.isclose(average_particle.mass, expected_average_mass, atol=1e-35 * u.kg)
+
+
+particle_multiplicities = [
+    {"e-": 1},
+    {"p+": 5, "e-": 11, "Fe-56 5+": 2},
+    {"p+": 4},
+    {CustomParticle(mass=1 * u.kg, charge=1 * u.C): 1},
+    {"p+": 5, CustomParticle(mass=1 * u.kg, charge=1 * u.C): 1},
+    {CustomParticle(): 1},
```

Comment:
@Quettle — thank you again for making the change in #1216.  I ended up having a chance to use it right away!

Reply:
Great I'm glad that I could help! By the way I was thinking of making something a bit similar than this PR. As I need custom particles to act as simple molecules in a simulation, I wanted to make an easy way to build them, maybe from ParticleLists.
For example `ParticleList([Particle("I"),Particle("I +")])` would give a` CustomParticle(mass = 2 * Particle("I").mass, charge = e.si, symbol = "I2 +")` What would you think of this idea?

Reply:
Interesting!  A challenge is that the symbols for `CustomParticle` objects are arbitrary and up to the user.  At the same time, it would also be really helpful to be able to create molecules...not only from `CustomParticle` objects but also from `Particle` objects.  I wonder if it would be worth trying out a prototype or two to figure out what the best way would be to make this work.  Thanks!

Cross-reference: #1026

Reply:
`ParticleList(anything)` should probably always return a `ParticleList`, without special casing. I think having a method do that conversion is cleaner, and doesn't add much boilerplate for the end user.